src/sinks/azure_common/service.rs

```diff
@@ -10,7 +10,7 @@ use futures::{future::BoxFuture, TryFutureExt};
 use tower::Service;
 use tracing::Instrument;
 use vector_common::internal_event::{
-    ByteSize, BytesSent, InternalEventHandle, Protocol, Registered,
+    ByteSize, BytesSent, InternalEventHandle as _, Protocol, Registered,
```

Comment:
Hmm, why do we need this trait import? My assumption would be that the generic trait bound on `vector_common::internal_event::register` would obviate the need for this. 🤔

Reply:
To expand, because the call to the trait methods is happening in the aforementioned method, that's why it doesn't make sense to me to need the trait in scope in places where we're just using `register!`.

Reply:
We are calling `InternalEventHandle::emit` in the components, which is why the trait import is needed. Your logic is sound, but for the `RegisterInternalEvent` trait.

Reply:
Oho! Alright, yeah, that tracks. 👍🏻